### PR TITLE
Add `parachain-staking` pallet

### PR DESCRIPTION
**Pull request by dastanfv**
- fork of `parachain-staking` pallet added to `pallets`. No modifications, copied as it is

### CODE REVIEW

**Comment by dastanfv** [comment]
maybe this shouldn't be added to runtime yet, since we do the upgrade at the end of the sprint

**Review by magecnion — APPROVED** [review]

File: ownership-chain/pallets/parachain-staking/src/traits.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use sp_runtime::DispatchErrorWithPostInfo;
+
+pub trait OnCollatorPayout<AccountId, Balance> {
+	fn on_collator_payout(
+		for_round: crate::RoundIndex,
+		collator_id: AccountId,
+		amount: Balance,
+	) -> Weight;
+}
+impl<AccountId, Balance> OnCollatorPayout<AccountId, Balance> for () {
```
Comment (asiniscalchi) [inline ownership-chain/pallets/parachain-staking/src/traits.rs:30]: any reason why there's no black space here ?

File: ownership-chain/pallets/parachain-staking/src/traits.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		_amount: Balance,
+	) -> Weight {
+		Weight::zero()
+	}
+}
+
+pub trait OnNewRound {
+	fn on_new_round(round_index: crate::RoundIndex) -> Weight;
+}
+impl OnNewRound for () {
```
Comment (asiniscalchi) [inline ownership-chain/pallets/parachain-staking/src/traits.rs:43]: any reason why there's no black space here ?

**Review by asiniscalchi — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -242,13 +242,18 @@ orml-traits = { git = "https://github.com/open-web3-stack/open-runtime-module-li
 orml-xcm-support = { git = "https://github.com/open-web3-stack/open-runtime-module-library", branch = "polkadot-v1.1.0", default-features = false }
 orml-xtokens = { git = "https://github.com/open-web3-stack/open-runtime-module-library", branch = "polkadot-v1.1.0", default-features = false }
 
+# Other pallets
+# (wasm)
+substrate-fixed = { git = "https://github.com/encointer/substrate-fixed", default-features = false }
```
Comment (asiniscalchi) [inline Cargo.toml:247]: what's this dependency for ?

**Review by asiniscalchi — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 3 above)
Comment (dastanfv) [reply Cargo.toml:247]: crate for fixed point arithmetic used by the `parachain-staking` pallet

File: ownership-chain/pallets/parachain-staking/src/traits.rs (hunk 1 above)
Comment (dastanfv) [reply ownership-chain/pallets/parachain-staking/src/traits.rs:30]: do you mean _blank_ space? if yes, as I said, the code is copied as it is, no modifications from my side.

File: Cargo.toml (hunk 3 above)
Comment (asiniscalchi) [reply Cargo.toml:247]: it's published on crates.io : https://crates.io/crates/substrate-fixed

Comment (asiniscalchi) [reply Cargo.toml:247]: https://github.com/encointer/substrate-fixed/pull/20

**Review by asiniscalchi — APPROVED** [review]

**Pull request merged**